docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -228,8 +228,160 @@ The table below summarizes the supported annotations:
+method is only unavailable for anonymous users.
+
+* #paramName == authentication.principal.username
+
+This syntax allows users to check if a parameter (or a field of the parameter) of the secured method is equal to the logged in username.
+
+Examples of this use case are:
+
+[source,java]
+----
+public class Person {
+
+    private final String name;
+
+    public Person(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+}
+
+@Component
+public class MyComponent {
+
+    @PreAuthorize("#username == authentication.principal.username") <1>
+    public void doSomething(String username, String other){
+
+    }
+
+    @PreAuthorize("#person.name == authentication.principal.username") <2>
+    public void doSomethingElse(Person person){
+
+    }
+}
+----
+
+<1> `doSomething` can be executed if the current logged in user is the same as the `username` method parameter
+<2> `doSomethingElse` can be executed if the current logged in user is the same as the as the `name` field of `person` method parameter
+
+* @beanName.method()
+
+This syntax allows developers to specify that the execution of method of a specific bean will determine if the current user can access the secured method.
+The syntax is best explained with an example.
+
+Let's assume that a `MyComponent` bean has been created like so:
+
+[source,java]
+----
+@Component
+public class MyComponent {
+
+    @PreAuthorize("@personChecker.check(#person, authentication.principal.username)")
+    public void doSomething(Person person){
+
+    }
+}
+----
+
+The `doSomething` method has been annotated with `@PreAuthorize` using an expression that indicates that method `check` of a bean named `personChecker` needs
+to invoked to determine whether the current user is authorized to invoke the `doSomething` method.
+
+An example of the PersonChecker could be:
+
+[source,java]
+----
+@Component
+public class PersonChecker {
+
+    @Override
+    public boolean check(Person person, String username) {
+        return person.getName().equals(username);
+    }
+}
+----
+
+Note that for the `check` method the parameter types much match what is specified in `@PreAuthorize` and that the return type must be a `boolean`.
+
+===== Combining expressions
+
+The `@PreAuthorize` annotations allows for the combination of expressions using logical `AND` / `OR`. Currently there is a limitation where only a single
+logical operation can be used (meaning mixing `AND` and `OR` isn't allowed).
+
+Some examples of allowed expressions are:
+
+[source,java]
+----
+
+    @PreAuthorize("hasAnyRole('user', 'admin') AND #user == principal.username")
+    public void allowedForUser(String user) {
+
+    }
+
+    @PreAuthorize("hasRole('user') OR hasRole('admin')")
+    public void allowedForUserOrAdmin() {
+
+    }
+
+    @PreAuthorize("hasAnyRole('view1', 'view2') OR isAnonymous() OR hasRole('test')")
+    public void allowedForAdminOrAnonymous() {
+        
+    }
+----
```

Comment:
Any chance you could add a warning/note about the precedence/parenthesis support?

Reply:
This was done

Reply:
I somehow missed it. I tweaked it a bit, could you check it's OK for you?

Reply:
Sure, I'll give it a check first thing in the morning

Reply:
Seems like you found it and updated it , :+1:

Reply:
Is there anything else or should we merge this?